drivers/hmis/app/models/hmis/ce/referral.rb

```diff
@@ -65,6 +71,45 @@ class Referral < GrdaWarehouseBase
       joins(:steps).merge(Hmis::WorkflowExecution::Step.where(id: assigned_step_ids).excluding_unavailable)
     end
 
+    # Referrals where there are completed steps assigned to swimlanes that the provided user participates in.
+    # These referrals are included in visibility for users that can view "own" referrals.
+    #
+    # This allows users to see referrals they're involved with through swimlane
+    # participation, even if no steps are currently directly assigned to them.
+    #
+    # Note: this logic is mirrored by `CeReferralAssignmentLoader#workflow_instance_ids_with_completed_swimlane_steps`
+    scope :with_completed_steps_assigned_to_swimlane, ->(user) do
+      # Get all referral participants for this user, with their workflow instance IDs
+      participants = user.ce_referral_participants.
+        joins(:referral).
+        select(:referral_id, :swimlane_id, 'ce_referrals.workflow_instance_id').
+        distinct
+
+      # Get all unique workflow instance IDs
+      instance_ids = participants.pluck(:workflow_instance_id).compact.uniq
```

Comment:
Should this `return none if instance_ids.empty?`

Reply:
Yes, updated